argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,207 @@
+    some pixels will not have valid disparity values. In those cases, we interpolate the predicted disparity map using
+    a simple nearest neighbor interpolation as in the KITTI Stereo 2015 benchmark [1] to assure we can compare it to
+    our semi-dense ground-truth disparity map. Current deep stereo matching methods normally predict disparity maps
+    with 100% density. Thus, an interpolation step is not needed for the evaluation.
+
+    The disparity errors metrics are the following:
+
+    1. all: Percentage of stereo disparity errors averaged over all ground truth pixels in the reference frame
+            (left stereo image).
+    2. bg: Percentage of stereo disparity errors averaged only over background regions.
+    3. fg: Percentage of stereo disparity errors averaged only over foreground regions.
+
+    The * (asterisk) means that the evaluation is performed using only the algorithm predicted disparities.
+    Even though the disparities might be sparse, they are not interpolated.
+
+    [1] http://www.cvlibs.net/datasets/kitti/eval_scene_flow.php?benchmark=stereo.
+
+Results:
+
+    We evaluate all metrics using three error thresholds: 3, 5, or 10 pixels.
+    The summary results are represented as a table containing the following fields:
+
+        all:10,
+        fg:10,
+        bg:10,
+        all*:10,
+        fg*:10,
+        bg*:10,
+        all:5,
+        fg:5,
+        bg:5,
+        all*:5,
+        fg*:5,
+        bg*:5,
+        all:3,
+        fg:3,
+        bg:3,
+        all*:3,
+        fg*:3,
+        bg*:3.
+
+    Note: The `evaluate` function will use all available logical cores on the machine.
+
+"""
+import argparse
+import logging
+import os
+from multiprocessing import Pool
+from pathlib import Path
+from typing import Dict, List
+
+import pandas as pd
+
+from argoverse.evaluation.stereo.constants import DEFAULT_ABS_ERROR_THRESHOLDS, DEFAULT_REL_ERROR_THRESHOLDS
+from argoverse.evaluation.stereo.utils import compute_disparity_error
+
+logger = logging.getLogger(__name__)
+
+
+class StereoEvaluator:
+    """Instantiates a StereoEvaluator object for evaluation."""
+
+    def __init__(
+        self,
+        pred_root_fpath: Path,
+        gt_root_fpath: Path,
+        figs_fpath: Path,
+        abs_error_thresholds: List[int] = DEFAULT_ABS_ERROR_THRESHOLDS,
+        rel_error_thresholds: List[float] = DEFAULT_REL_ERROR_THRESHOLDS,
+        save_disparity_error_image: bool = False,
+        num_procs: int = -1,
+    ) -> None:
+        """
+        Args:
+            pred_fpath_root: Path to the folder which contains the stereo predictions.
+            gt_fpath_root: Path to the folder which contains the stereo ground truth.
+            figs_fpath: Path to the folder which will contain the output figures.
+            abs_error_thresholds: List with the absolute disparity error thresholds.
+            rel_error_thresholds: List with the relative disparity error thresholds.
```

Comment:
i think the List part is already clear from the function signature/type hint, so I would say something a bit more descriptive like
```python
abs_error_thresholds: Absolute disparity error thresholds, in pixels
```